apps/server/src/infra/cache/cache-store.factory.ts

```diff
@@ -0,0 +1,48 @@
+import { LegacyLogger } from '@core/logger';
+import { ValkeyConfig, ValkeyFactory } from '@infra/valkey-client';
+import KeyvValkey from '@keyv/valkey';
+import { ConfigService } from '@nestjs/config';
+import Keyv from 'keyv';
+import { CacheConfig } from './interface';
+import { KeyvValkeyAdapter } from './keyv-valkey.adapter';
+
+export class CacheStoreFactory {
+	private static logger: LegacyLogger;
+	private static config: ValkeyConfig;
+
+	public static async build(
+		configService: ConfigService<CacheConfig>,
+		logger: LegacyLogger
```

Comment:
Hier wird ein logger rein gereicht und das sollte auch reichen. 
this.logger macht denke ich kein Sinn. Was aber auf keinen Fall passieren darf, ist das der context des loggers hier überschrieben wird.